crates/miden-tx/src/tests/kernel_tests/test_fpi.rs

```diff
@@ -839,40 +879,339 @@ fn test_simple_nested_fpi() {
         .unwrap();
 }
 
+/// Test that code will panic in attempt to create more than 63 foreign accounts.
+///
+/// Attempt to create a 64th foreign account first triggers the assert during the account data
+/// loading, but we have an additional assert during the account stack push just in case.
 #[test]
 fn test_nested_fpi_stack_overflow() {
-    let mut foreign_accounts = Vec::new();
+    // use a custom thread to increase its stack capacity
+    std::thread::Builder::new().stack_size(10 * 1_048_576).spawn(||{
```

Comment:
Not very important, but I still wonder what takes up so much stack space here - any idea?

Reply:
Probably because of the loaded account data. An account requires 8192 elements in memory, which is 64 KB for each account, so for 64 accounts it's already 4 MB. To make this rest not to fail with stack overflow we need to create a thread with at least 8 MB of memory, so probably other parts of the program take other 4 MB.

Reply:
Mh, this is the Rust stack, right, so it shouldn't have anything to do with the VM. The VM memory is on the heap, not the stack, so I don't think it should be related to the account memory. Anyway, probably not important enough to investigate.